cpp/src/arrow/c/bridge_test.cc

```diff
@@ -31,6 +31,7 @@
 #include "arrow/c/bridge.h"
 #include "arrow/c/helpers.h"
 #include "arrow/c/util_internal.h"
+#include "arrow/compute/api_vector.h"
```

Comment:
I don't think we depend on ARROW_COMPUTE currently for these tests. We should at least guard against it (use a `#ifdef ARROW_COMPUTE` and skip the test otherwise).

Reply:
Or instead we could have a `REEArrayFromJSON` helper in https://github.com/apache/arrow/blob/main/cpp/src/arrow/ipc/json_simple.h that, just like `DictArrayFromJSON`, would take separate strings for run ends and values.

Reply:
`RunEndEncode` and decode are considered one of the basic kernels that don't need the full arrow-compute build to be available as they are so integral to doing anything with REEs. 

> ...would take separate strings for run ends and values.

That makes writing these tests so much harder and the code less readable. Can I create an issue about this to be fixed outside the context of this PR? I could make `RunEndEncode` usable without the entirety of `api_vector.h` --- as a C++ API utility function that is shared between the compute kernel and internal code like the test utilities?

Reply:
I will add the ifdefs and mention the issue in the code https://github.com/apache/arrow/issues/37221